packages/desktop-client/src/components/reports/Reports.js

```diff
@@ -0,0 +1,454 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { connect } from 'react-redux';
+
+import * as d from 'date-fns';
+import { bindActionCreators } from 'redux';
+
+import * as actions from 'loot-core/src/client/actions';
+import { send } from 'loot-core/src/platform/client/fetch';
+import * as monthUtils from 'loot-core/src/shared/months';
+import { View, P } from 'loot-design/src/components/common';
+import { colors, styles } from 'loot-design/src/style';
+
+import { ChooseChart, ChooseChartHeader, ChartExtraColumn } from './Charts';
+import Header from './Header';
+import { HeaderReport } from './Header';
+import { masterDataSpreadsheet } from './spreadsheets/master-spreadsheet';
+import useReport from './useReport';
+
+function AllReports({ categories }) {
+  const [filterz, setFilterz] = useState([]);
+  const [filt, setFilt] = useState([]);
+  const [disableFilter, setDisableFilter] = useState(true);
+  const [isCashFlow, setIsCashFlow] = useState(true);
+  const [isIE, setIsIE] = useState(false);
+  const [isTotals, setIsTotals] = useState(false);
+  const [isTrends, setIsTrends] = useState(false);
+  const [isNetWorth, setIsNetWorth] = useState(false);
```

Comment:
Can you replace these with a single state variable that uniquely represents the current screen? (such that all possible states represent a valid UI state). For example, right now theoretically all of the boolean states here could be `true` but that wouldn’t correspond to a valid UI state

Reply:
Done. Had to use one state for the main reporting links and a different one for the "totals/trends" within Income&Expenses view.